main.go

```diff
@@ -1000,16 +1000,48 @@ func downloadFromS3WithPrefix(client *minio.Client, prefix, bucket string) (stri
 	}
 	// if folder is included, strip it so it doesnt end up in a folder on the host itself
 	targetFilename := path.Base(filename)
+	targetFileLocation := fmt.Sprintf("%s/%s", backupBaseDir, targetFilename)
 	for retries := 0; retries <= s3ServerRetries; retries++ {
-		err := client.FGetObject(bucket, filename, backupBaseDir+"/"+targetFilename, minio.GetObjectOptions{})
+		object, err := client.GetObject(bucket, filename, minio.GetObjectOptions{})
 		if err != nil {
 			log.Infof("Failed to download etcd snapshot file [%s]: %v, retried %d times", filename, err, retries)
 			if retries >= s3ServerRetries {
 				log.Warningf("Failed to download etcd snapshot file [%s]: %v", filename, err)
 				break
 			}
 		}
+		defer object.Close()
+
+		localFile, err := os.Create(targetFileLocation)
```

Comment:
are you sure you want all the below (file create, stat, copy) to be inside of this retry loop? It was originally intended for retrying minio with some network failure or something. I think the retry loop should just get the minio object and then do all the file stuff afterwards.